sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/EventGridPublisherAsyncClient.java

```diff
@@ -70,7 +76,8 @@ public EventGridServiceVersion getServiceVersion() {
         return Flux.fromIterable(events)
             .map(EventGridEvent::toImpl)
             .collectList()
-            .flatMap(list -> this.impl.publishEventsAsync(this.hostname, list, context));
+            .flatMap(list -> this.impl.publishEventsAsync(this.hostname,
+                list, context.addData(AZ_TRACING_NAMESPACE_KEY, Constants.EVENT_GRID_TRACING_NAMESPACE_VALUE)));
```

Comment:
This internal api is called by a public API, which calls `FluxUtil.withContext` to create a Context. So I assume the context won't be null.
In debugging, I see it's an empty Context instance instead of null.

Reply:
@YijunXieMS - this method is also called from sync client and the user can pass a null context.

User can call `sendEventsWithResponse(events, null)`
```java
    public Response<Void> sendEventsWithResponse(Iterable<EventGridEvent> events, Context context) {
        return asyncClient.sendEventsWithResponse(events, context).block();
    }
```